ultimarc/devices/_mappings.py

```diff
@@ -0,0 +1,517 @@
+#
+PREIPACMapping = {
+    "A": 0x1C,
+    "B": 0x32,
+    "C": 0x21,
+    "D": 0x23,
+    "E": 0x24,
+    "F": 0x2B,
+    "G": 0x34,
+    "H": 0x33,
+    "I": 0x43,
+    "J": 0x3B,
+    "K": 0x42,
+    "L": 0x4B,
+    "M": 0x3A,
+    "N": 0x31,
+    "O": 0x44,
+    "P": 0x4D,
+    "Q": 0x15,
+    "R": 0x2D,
+    "S": 0x1B,
+    "T": 0x2C,
+    "U": 0x3C,
+    "V": 0x2A,
+    "W": 0x1D,
+    "X": 0x22,
+    "Y": 0x35,
+    "Z": 0x1A,
+    "1": 0x16,
+    "2": 0x1E,
+    "3": 0x26,
+    "4": 0x25,
+    "5": 0x2E,
+    "6": 0x36,
+    "7": 0x3D,
+    "8": 0x3E,
+    "9": 0x46,
+    "0": 0x45,
+    "ESC": 0x76,
+    "-": 0x4E,
+    "=": 0x55,
+    "BKSP": 0x66,
+    "TAB": 0x0D,
+    "[": 0x54,
+    "]": 0x5B,
+    "ENTER": 0x5A,
+    "KP ENTER": 0x5A,
+    "CTRL L": 0x14,
+    "CTRL R": 0x14,
+    ";": 0x4C,
+    "'": 0x52,
+    "`": 0x0E,
+    "SHIFT L": 0x12,
+    "\\": 0x5D,
+    ",": 0x41,
+    ".": 0x49,
+    "/": 0x4A,
+    "KP /": 0x4A,
+    "SHIFT R": 0x59,
+    "PRNT SCRN": 0x7C,
+    "ALT L": 0x11,
+    "ALT R": 0x11,
+    "SPACE": 0x29,
+    "CAPS": 0x58,
+    "F1": 0x05,
+    "F2": 0x06,
+    "F3": 0x04,
+    "F4": 0x0C,
+    "F5": 0x03,
+    "F6": 0x0B,
+    "F7": 0x83,
+    "F8": 0x0A,
+    "F9": 0x01,
+    "F10": 0x09,
+    "F11": 0x78,
+    "F12": 0x07,
+    "NUM": 0x77,
+    "SCROLL": 0x7E,
+    "KP 1": 0x69,
+    "KP 2": 0x72,
+    "KP 3": 0x7A,
+    "KP 4": 0x6B,
+    "KP 5": 0x73,
+    "KP 6": 0x74,
+    "KP 7": 0x6C,
+    "KP 8": 0x75,
+    "KP 9": 0x7D,
+    "KP 0": 0x70,
+    "KP /": 0x4A,
+    "KP *": 0x7C,
+    "KP -": 0x7B,
+    "KP +": 0x79,
+    "KP .": 0x71,
+    "SYSRQ": 0x84,
+    "WIN MENU": 0x2F,
+    "up": 0xF5,
+    "down": 0xF2,
+    "right": 0xF4,
+    "left": 0xEB,
+    "1right": 0x02,
+    "3right": 0x02,
+    "1up": 0x01,
+    "3up": 0x01,
+    "1down": 0x06,
+    "3down": 0x06,
+    "1left": 0x04,
+    "3left": 0x04,
+    "1sw1": 0x03,
+    "3sw1": 0x03,
+    "1sw2": 0x08,
+    "3sw2": 0x08,
+    "1sw3": 0x05,
+    "3sw3": 0x05,
+    "1sw4": 0x0A,
+    "3sw4": 0x0A,
+    "1sw5": 0x07,
+    "3sw5": 0x07,
+    "1sw6": 0x0C,
+    "3sw6": 0x0C,
+    "1sw7": 0x18,
+    "3sw7": 0x18,
+    "1sw8": 0x1A,
+    "3sw8": 0x1A,
+    "1start": 0x10,
+    "3start": 0x10,
+    "1coin": 0x14,
+    "3coin": 0x14,
+    "2right": 0x09,
+    "4right": 0x09,
+    "2up": 0x0D,
+    "4up": 0x0D,
+    "2down": 0x0E,
+    "4down": 0x0E,
+    "2left": 0x0B,
+    "4left": 0x0B,
+    "2sw1": 0x0F,
+    "4sw1": 0x0F,
+    "2sw2": 0x11,
+    "4sw2": 0x11,
+    "2sw3": 0x13,
+    "4sw3": 0x13,
+    "2sw4": 0x15,
+    "4sw4": 0x15,
+    "2sw5": 0x17,
+    "4sw5": 0x17,
+    "2sw6": 0x19,
+    "4sw6": 0x19,
+    "2sw7": 0x1B,
+    "4sw7": 0x1B,
+    "2sw8": 0x1C,
+    "4sw8": 0x1C,
+    "2start": 0x12,
+    "4start": 0x12,
+    "2coin": 0x16,
+    "4coin": 0x16,
+    "1a": 0x1D,
+    "1b": 0x1E,
+    "2a": 0x1F,
+    "2b": 0x20,
+    "m1": 0xe0,
+    "m2": 0xe1,
+    "m3": 0xe2,
+    "m4": 0xe3,
+}
+
+KeyLookupTable = (
```

Comment:
I am not sure I want to have all the boards information in one location this time.   Having it here for reference while the boards are being built out is good.  I was invisioning having each board contain its own information about where data is placed.

Reply:
That sounds good to me.